phac-nml/irida-next
# Sample Exports: Format Filtering UI

## What does this PR do and why?
This PR adds the UI to filter sample export files by format.

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/3696c6d4-7b88-4b48-97ee-b84011f5f368)

## How to set up and validate locally
1. Through the UI, navigate to a group or project samples table and select samples with attachments of multiple formats
2. Click the `Create Export` dropdown and select `Sample Export`
3. In the dialog, verify the following:
- The sample accordion header contains `Samples (X selected)`. Ensure the correct number of samples is displayed
- Open the accordion and ensure the export description/explanation and sample listing are present
- Under file formats, use the `Add all` and `Remove all` buttons. Ensure they are disabled after use. Ensure the `Submit` button is also disabled when the `Selected` list contains no values.
4. Submit the export with a format(s), ensuring you're excluding/not selecting certain formats contained by the selected samples' attachments
5. After 30s on the export's `show` page, download the export and ensure the export files, the `manifest.json` contents, and the `Preview` tab only contain the expected files (ie: formats that were not selected should not be present).
6. Repeat the above with all formats and formats the samples do not have. Ensure all files and no files are included in the export, respectively.

Other tests:
7. Start a `Linelist Export`, and ensure the sample accordion header contains `Samples (X selected)` and that the correct number of samples is displayed. The export description was also moved into the accordion. Open the accordion and verify the description is present.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
